app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -837,6 +852,59 @@ class BrowserTabFragment :
         }
     }
 
+    private fun configureCustomTab() {
+        omnibar.omniBarContainer.hide()
+        omnibar.fireIconMenu.hide()
+        omnibar.tabsMenu.hide()
+
+        omnibar.toolbar.background = ColorDrawable(tabToolbarColor)
+        omnibar.toolbarContainer.background = ColorDrawable(tabToolbarColor)
+
+        omnibar.customTabToolbarContainer.customTabToolbar.show()
+
+        omnibar.customTabToolbarContainer.customTabCloseIcon.setOnClickListener {
+            requireActivity().finish()
+        }
+
+        omnibar.customTabToolbarContainer.customTabShieldIcon.setOnClickListener { _ ->
+            val params = PrivacyDashboardHybridScreen.PrivacyDashboardHybridWithTabIdParam(tabId)
+            val intent = globalActivityStarter.startIntent(requireContext(), params)
+            intent?.let { startActivity(it) }
+            pixel.fire(CustomTabPixelNames.CUSTOM_TABS_PRIVACY_DASHBOARD_OPENED)
+        }
+
+        omnibar.customTabToolbarContainer.customTabDomain.text = viewModel.url?.extractDomain()
+        omnibar.customTabToolbarContainer.customTabDomainOnly.text = viewModel.url?.extractDomain()
+        omnibar.customTabToolbarContainer.customTabDomainOnly.show()
+
+        val foregroundColor = calculateForegroundColor(tabToolbarColor)
+        omnibar.customTabToolbarContainer.customTabCloseIcon.setColorFilter(foregroundColor)
+        omnibar.customTabToolbarContainer.customTabDomain.setTextColor(foregroundColor)
+        omnibar.customTabToolbarContainer.customTabDomainOnly.setTextColor(foregroundColor)
+        omnibar.customTabToolbarContainer.customTabTitle.setTextColor(foregroundColor)
+        omnibar.browserMenuImageView.setColorFilter(foregroundColor)
+
+        requireActivity().window.navigationBarColor = tabToolbarColor
+        requireActivity().window.statusBarColor = tabToolbarColor
+    }
+
+    private fun calculateForegroundColor(color: Int): Int {
```

Comment:
this returns either black or white, but the function name suggests any color could be returned. how about making it more explicit?

Reply:
Thanks :+1:  Renamed.